api/app/signals/apps/api/generics/permissions/base.py

```diff
@@ -0,0 +1,118 @@
+from django.conf import settings
```

Comment:
This file might by split into a `base.py` and something like `actions.py`. Not all permissions herein are general or base.